src/System.Drawing.Primitives/src/System/Drawing/Rectangle.cs

```diff
@@ -0,0 +1,501 @@
+            return Contains(pt.X, pt.Y);
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Determines if the rectangular region represented by
+        ///    <paramref name="rect"/> is entirely contained within the rectangular region represented by 
+        ///       this <see cref='System.Drawing.Rectangle'/> .
+        ///    </para>
+        /// </devdoc>
+        [Pure]
+        public bool Contains(Rectangle rect)
+        {
+            return (X <= rect.X) &&
+            ((rect.X + rect.Width) <= (X + Width)) &&
+            (Y <= rect.Y) &&
+            ((rect.Y + rect.Height) <= (Y + Height));
+        }
+
+        public override int GetHashCode()
+        {
+            return unchecked((int)((UInt32)X ^
+                        (((UInt32)Y << 13) | ((UInt32)Y >> 19)) ^
+                        (((UInt32)Width << 26) | ((UInt32)Width >> 6)) ^
+                        (((UInt32)Height << 7) | ((UInt32)Height >> 25))));
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Inflates this <see cref='System.Drawing.Rectangle'/>
+        ///       by the specified amount.
+        ///    </para>
+        /// </devdoc>
+        public void Inflate(int width, int height)
+        {
+            X -= width;
+            Y -= height;
+            Width += 2 * width;
+            Height += 2 * height;
+        }
+
+        /// <devdoc>
+        ///    Inflates this <see cref='System.Drawing.Rectangle'/> by the specified amount.
+        /// </devdoc>
+        public void Inflate(Size size)
+        {
+
+            Inflate(size.Width, size.Height);
+        }
+
+        /// <devdoc>
+        ///    <para>
+        ///       Creates a <see cref='System.Drawing.Rectangle'/>
+        ///       that is inflated by the specified amount.
+        ///    </para>
+        /// </devdoc>
+        public static Rectangle Inflate(Rectangle rect, int x, int y)
+        {
+            Rectangle r = rect;
+            r.Inflate(x, y);
+            return r;
+        }
+
+        /// <devdoc> Creates a Rectangle that represents the intersection between this Rectangle and rect.
+        /// </devdoc>
+        public void Intersect(Rectangle rect)
+        {
+            Rectangle result = Rectangle.Intersect(rect, this);
+
+            X = result.X;
+            Y = result.Y;
+            Width = result.Width;
+            Height = result.Height;
+        }
+
+        /// <devdoc>
+        ///    Creates a rectangle that represents the intersetion between a and
+        ///    b. If there is no intersection, null is returned.
+        /// </devdoc>
+        public static Rectangle Intersect(Rectangle a, Rectangle b)
+        {
+            int x1 = Math.Max(a.X, b.X);
+            int x2 = Math.Min(a.X + a.Width, b.X + b.Width);
+            int y1 = Math.Max(a.Y, b.Y);
+            int y2 = Math.Min(a.Y + a.Height, b.Y + b.Height);
+
+            if (x2 >= x1
+                && y2 >= y1)
+            {
+
```

Comment:
Nit: unnecessary blank line, and the above if conditions can all be on the same line